src/Collection.php

```diff
@@ -353,14 +353,14 @@ public static function fromCallable(callable $callable, iterable $parameters = [
     }
 
     /**
-     * @param null|non-negative-int $length
+     * @param Closure(resource): mixed $consumer
      *
-     * @return CollectionInterface<int, string>&self<int, string>
+     * @return CollectionInterface<int, string|mixed>&self<int, string|mixed>
      */
-    public static function fromFile(string $filepath, ?int $length = 2): CollectionInterface
```

Comment:
hmm we could keep the `$length` parameter as well with a default value, I'm not really sure why should it be removed? Then the change is not breaking BC at all :).

And when you don't want to use it you can still call the constructor using the named argument:

```php
Collection::fromFile('data.csv', consumer: fgetcsv(...));
```

Reply:
I decided to remove it since it can be put back through the `$consumer` as such:

```php
Collection::fromFile('data.csv', static fn ($resource) => fgets($resource, 3))
```

Therefore, it's not needed any more since we can now customize the `$consumer`.

Reply:
fair enough, I guess you have more control overall now, let's go with it 👍

Reply:
And the thing is that I removed it from `loophp/iterators` already... https://github.com/loophp/iterators/blob/main/src/ResourceIteratorAggregate.php